doc/guides/modules.rst

```diff
@@ -32,6 +32,21 @@ Zephyr depends on several categories of modules, including but not limited to:
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+Zephyr also supports extension of functionality with out-of-tree modules.
+These are modules downstream of the Zephyr repository and may be managed
+outside of the Zephyr Project.  Out-of-tree modules will likely be their
```

Comment:
This "will likely be" is your use case, I know, but I don't think it's appropriate for this documentation, unless you have some hard data showing that it is most likely.